segmentstore/server/src/main/java/io/pravega/segmentstore/server/logs/DataFrameReader.java

```diff
@@ -45,6 +49,10 @@
     @Setter (AccessLevel.PROTECTED)
     private long lastReadSequenceNumber;
     private boolean closed;
+    private final String traceObjectId;
+    @Getter (AccessLevel.PACKAGE)
+    private final AtomicInteger maxOverlapToCheckForDuplicates = new AtomicInteger(0); // Default, no tolerance to repeated elements.
```

Comment:
Why is this mutable and not just a constant? If it's for testing perhaps a better way would be to make a method as a getter and override it in the test.

Reply:
Fixed: set this as a final int set via constructor.